### PR TITLE
ghdl: 1.0.0 -> 2.0.0

### PR DESCRIPTION
**Pull request by doronbehar**
###### Motivation for this change

ghdl was broken ever since https://github.com/NixOS/nixpkgs/issues/186482 .

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by risicle** [comment]
@ofborg build yosys-ghdl

fails for me, nixos x86_64

**Comment by doronbehar** [comment]
`yosys-ghdl` builds now, not sure if functional though. My motivation to update `ghdl` is for a new package that needs it.

File: pkgs/development/compilers/ghdl/default.nix [hunk 1]
```diff
@@ -1,40 +1,62 @@
...
   };
 
   patches = [
-    # Allow compilation with GNAT 11, picked from master
+    # https://github.com/ghdl/ghdl/issues/2056
     (fetchpatch {
-      name = "fix-gnat-11-compilation.patch";
-      url = "https://github.com/ghdl/ghdl/commit/8356ea3bb4e8d0e5ad8638c3d50914b64fc360ec.patch";
-      sha256 = "04pzn8g7xha8000wbjjmry6h1grfqyn3bjvj47hi4qwgl21wfjra";
+      url = "https://github.com/ghdl/ghdl/commit/f8b87697e8b893b6293ebbfc34670c32bfb49397.patch";
```
Comment (risicle) [inline pkgs/development/compilers/ghdl/default.nix:28]:
```suggestion
      name = "fix-build-gcc-12.patch";
      url = "https://github.com/ghdl/ghdl/commit/f8b87697e8b893b6293ebbfc34670c32bfb49397.patch";
```

**Review by risicle — APPROVED** [review]
`nixpkgs-review` happy, nixos x86_64.

(no other platforms seem relevant)

**Comment by doronbehar** [comment]
Thanks for the review, will merge when CI is green.

**Pull request merged**